src/game_api/sound_manager.hpp

```diff
@@ -260,6 +260,9 @@ struct SoundInfo
     std::string sound_name; // not 100% sure if it's standard
 };
 
+// there is actually base class that consists of up to the left/right channel, used for the sfx
+// then normal sounds that you can find in entities, screens itp. have up to the paddings
+// finally music in GameManages has the extra bool in BackgroundSound
```

Comment:
Should i split this? (if yes, would love some name recommendations)

Reply:
I guess it would be nice if it means more of those sound types exposed. But I don't really know about name, since now I guess we only have to use a name without altering the SoundMeta one...

Reply:
You can edit the name, as long as there is no `new` constructor exposed to lua, no one should really care about the metatable name

Reply:
Honestly, I can't come up with any name ideas, since I'm not completely sure what's the difference between those clases :cat2:.
Is the current SoundMeta for loop-able long sounds? Having a bool for `play_ending_secuence` makes it sound like it's repeatable and can be ended by game logic, and also the `start_over`

Reply:
I guess you could leave it for another PR if you wanted

Reply:
The `start_over` is more of a name i give it to make sense in the API rather then what actual functionality it holds.
The class name doesn't need to reflect much. In the source code it's probably just base class like "Sound".

That base class is used for stuff like: play and forget. You can just play sfx and not worry about it as the internals of it will deal with cleaning after it finishes etc.
Then `SoundMeda` is for sounds you want you either reuse or need to maintain (like keep updating the position etc.), it also used when you need to know if previous sound is still playing. Pretty sure you need to call the destroy function on it yourself

At the end is `BackgroundSound` which is probably only implemented for the main "music" in game_manager

There might be more that we don't know of, or that don't differ by the class members (only virtual overrides)

Reply:
some rough ideas, `ContinuousSound`, `LoopableSound`? I suppose some examples could be the UFO laser shoot, lamassu laser prepare sound or Lamahu tentacles.
Do you have any ideas for the name yourself?

Reply:
SFX -> Sound -> BackgroundSound  ?

Reply:
Sounds pretty good
It could also be named SoundEffect and Sound changed into something like DynamicSound/ActiveSound to give contrast with the base class I guess
Also, is this gonna continue on this PR or should I approve this after testing everything works?

Reply:
> Also, is this gonna continue on this PR or should I approve this after testing everything works?

We can move this conversation to discord and be done with this PR i guess

Reply:
Alright